Size.pm

```diff
@@ -0,0 +1,68 @@
+package App::Ack::Filter::Size;
+
+use strict;
+use warnings;
+use base 'App::Ack::Filter';
+
+use Carp 'croak';
+
+use App::Ack::Filter ();
+
+sub _parse_size {
+    my $s = $_[0] || return 0;
+
+    if ( $s =~ m/^\s*(\d+(?:\.\d+)?)(?:\s*([KMGT]?)B?)?\s*$/i ) {
+        my $n = $1;
+        if ($2) {
+            my $u = lc $2;
+            $n *= 1024 while $u =~ tr/tgmk/gmk/d;
+        }
+        return int $n;
+    }
+    else {
+        Carp::croak('Invalid size');
+    }
+}
+
+sub new {
+    my ( $class, $min, $max ) = @_;
+    return bless {
+        min => $min,
+        max => $max,
+    }, $class;
+}
+
+sub filter {
+    my ( $self, $resource ) = @_;
+
+    my $min = $self->{'min'} || 0;
+    my $max = $self->{'max'};
+
+    my $file = $resource->name;
+
+    return 1 if $file eq '-';
+
+    my $size = (-s $file) || 0; # paranoid?
+
+    return 0 if $max and $size > $max;
+    return $size >= $min;
+}
+
+sub inspect {
+    my ( $self ) = @_;
+
+    my $min = $self->{'min'} || 0;
+    my $max = $self->{'max'} || '*';
+
+    return ref($self) . " - $min..$max";
+}
+
+sub to_string {
+    shift->inspect;
+}
+
+BEGIN {
+    App::Ack::Filter->register_filter(size => __PACKAGE__);
```

Comment:
I don't know if it makes sense to register this filter; registered filters are used for `--type-add` and friends.  Do we want users to be able to say `--type-add=big-file:size:0,1000000`?

If we're not going to use this as a registered filter, I don't know if it makes sense to implement this feature using a filter class; users using the feature are going to incur extra overhead from a method call, which really adds up in large codebases.